File: Tier1/Shared/DTOs/RefugeeUpdateDTO.cs
namespace Shared.DTOs;

public class RefugeeUpdateDTO
{
    public string Email { get; }
    public string? Password { get; }
    public char? Gender { get; }
    public string? Nationality { get; }
    public string? FirstName { get; }
    public string? MiddleName { get; }
    public string? LastName { get; }
    public int? FamilySize { get; }
    public string? Description { get; }

    public RefugeeUpdateDTO(string email, string? password, char? gender, string? nationality, string? firstName, string? middleName, string? lastName, int? familySize, string? description)
    {
        Email = email;
        Password = password;
        Gender = gender;
        Nationality = nationality;
        FirstName = firstName;
        MiddleName = middleName;
        LastName = lastName;
        FamilySize = familySize;
        Description = description;
    }
}

File: Tier2/src/main/java/via/sep3/group11/tier2/shared/DTOs/RefugeeUpdateDTO.java
package via.sep3.group11.tier2.shared.DTOs;

public class RefugeeUpdateDTO {
    private String email;
    private String password;
    private char gender;
    private String nationality;
    private String firstName;
    private String middleName;
    private String lastName;
    private int familySize;
    private String description;

    public RefugeeUpdateDTO(String email, String password, char gender, String nationality, String firstName, String middleName, String lastName, int familySize, String description) {
        this.email = email;
        this.password = password;
        this.gender = gender;
        this.nationality = nationality;
        this.firstName = firstName;
        this.middleName = middleName;
        this.lastName = lastName;
        this.familySize = familySize;
        this.description = description;
    }

    public String getEmail() {
        return email;
    }

    public String getPassword() {
        return password;
    }

    public char getGender() {
        return gender;
    }

    public String getNationality() {
        return nationality;
    }

    public String getFirstName() {
        return firstName;
    }

    public String getMiddleName() {
        return middleName;
    }

    public String getLastName() {
        return lastName;
    }

    public int getFamilySize() {
        return familySize;
    }

    public String getDescription() {
        return description;
    }
}
